sbc_compassion/models/import_letters_history.py

```diff
@@ -192,121 +162,172 @@ def _run_analyze(self):
+                            # write data in tempfile
+                            zip_file.write(base64.b64decode(
+                                attachment.with_context(
+                                    bin_size=False).datas))
+                            zip_file.flush()
+
+                            zip_ = zipfile.ZipFile(
+                                zip_file, 'r')
+                            # loop over files inside zip
+                            directory = tempfile.mkdtemp()
+                            for f in zip_.namelist():
+                                    zip_.extract(
+                                        f, directory)
+                                    absname = directory + \
+                                        '/' + f
+                                    if os.path.isfile(absname):
+                                        # remove if PDF is working
+                                        if func.isPDF(absname):
+                                            raise exceptions.Warning(
+                                                _("PDF not implemented yet"))
+                                        filename = f.split('/')[-1]
+                                        inst._analyze_attachment(absname,
+                                                                 filename)
+                            shutil.rmtree(directory)
                     # case with normal format ([PDF,]TIFF)
                     elif func.check_file(attachment.name) == 1:
                         # remove if PDF is working
                         if func.isPDF(attachment.name):
                             raise exceptions.Warning(
                                 _("PDF not implemented yet"))
-                        inst._analyze_attachment(
-                            inst.path + str(attachment.name),
-                            attachment.datas)
+                        ext = os.path.splitext(attachment.name)[1]
+                        with tempfile.NamedTemporaryFile(
+                                suffix=ext) as file_:
+                            file_.write(base64.b64decode(
+                                attachment.with_context(
+                                    bin_size=False).datas))
+                            file_.flush()
+                            inst._analyze_attachment(file_.name,
+                                                     attachment.name)
                     else:
                         raise exceptions.Warning(
                             'Still a file in a non-accepted format')
                 else:
                     raise exceptions.Warning(_('Two files are the same'))
             # remove all the files (now there are inside import_line_ids)
-            for attachment in inst.data:
-                attachment.unlink()
 
-    def _analyze_attachment(self, file_, data=None):
+    def _analyze_attachment(self, file_, filename):
```

Comment:
I have created a method for the whole part about the QR code (with the rotation)